src/main/scala/temple/generate/docker/DockerFileGenerator.scala

```diff
@@ -0,0 +1,37 @@
+package temple.generate.docker
+
+import temple.generate.docker.ast.Statement._
+import temple.generate.docker.ast.{DockerfileRoot, Statement}
+
+/** Generator object for building Dockerfiles from the Dockerfile AST objects */
+object DockerFileGenerator {
+
+  /** Given a [ast.Statement], generate a valid string */
+  private def generateStatement(statement: Statement): String =
+    statement match {
+      case Env => "ENV"
+      case From(image, tag) =>
+        s"FROM $image" + (tag match {
+          case Some(tagString) => s":$tagString"
+          case None            => ""
+        })
```

Comment:
This would be equivalent to one of the following: I feel `mkSQL` should be renamed and moved if this were to happen
```scala
mkSQL("FROM", mkSQL.noSpaces(image, tag.map(":" + _))) // With a noSpaces method added
mkSQL("FROM", image, tag.map(":" + _)) // With colon set as a closing punctuation mark
```

Reply:
This is a good point - is there anything in `SQLTerm` that is actually specific to SQL generation? Could be moved to a general `generate.utils` package that does all sorts of helpful stuff for generation like this

Reply:
I don’t believe so. It should apply to any whitespace-independent language, though it is opinionated about when to insert spaces (not after `(` or whitespace, and not before `,;)` or whitespace). You'd need to add colon to that, and probably `[{`/`}]` too, as well as possibly adding a noSpaces method

Reply:
Also you think it's worth adding a whole JSON lib dependency for this? Scala doesn't have JSON processing built in